Gemfile.lock

```diff
@@ -1,42 +1,55 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    diff-lcs (1.2.4)
-    facter (1.7.2)
-    fakefs (0.4.3)
-    fuubar (1.1.1)
-      rspec (~> 2.0)
-      rspec-instafail (~> 0.2.0)
-      ruby-progressbar (~> 1.0)
-    multi_json (1.7.7)
-    rake (10.1.1)
-    rspec (2.13.0)
-      rspec-core (~> 2.13.0)
-      rspec-expectations (~> 2.13.0)
-      rspec-mocks (~> 2.13.0)
-    rspec-core (2.13.1)
-    rspec-expectations (2.13.0)
-      diff-lcs (>= 1.1.3, < 2.0)
-    rspec-instafail (0.2.4)
-    rspec-mocks (2.13.1)
-    ruby-progressbar (1.1.1)
-    simplecov (0.7.1)
-      multi_json (~> 1.0)
-      simplecov-html (~> 0.7.1)
-    simplecov-html (0.7.1)
-    thor (0.18.1)
+    CFPropertyList (3.0.0)
+    diff-lcs (1.3)
+    docile (1.3.2)
+    facter (2.5.1)
```

Comment:
What changes can we expect when going from facter 1.7.2 to 2.5.1? Anything not backwards compatible?

Reply:
Facter 2 introduced structured facts, which might require some manual investigation into how the Razor server handles deserializing and storing them. API-wise it also has some changes to support those structured facts - I don't know what we support as far as custom facts in the MK

Reply:
That is what I suspected. Are structured facts additions only, or is it breaking for legacy facts?

Reply:
All legacy facts will still exist, but I don't think there's an API to get *only* the legacy facts, so if anything downstream can't handle the structure we'll need to filter it appropriately